libraries/rush-lib/src/utilities/prompts/SearchListPrompt.ts

```diff
@@ -0,0 +1,300 @@
+      .pipe(takeUntil(events.line))
+      .forEach(this._onNumberKey.bind(this) as (evt: unknown) => void);
+
+    // eslint-disable-next-line no-void
+    void events.keypress
+      .pipe(takeUntil(validation.success))
+      .forEach(this._onKeyPress.bind(this) as (evt: unknown) => void);
+
+    this.render();
+    return this;
+  }
+
+  private _onUpKey(): void {
+    return this._adjustSelected(-1);
+  }
+
+  private _onDownKey(): void {
+    return this._adjustSelected(1);
+  }
+
+  private _onNumberKey(input: number): void {
+    if (input <= this.opt.choices.realLength) {
+      this._selected = input - 1;
+    }
+
+    this.render();
+  }
+
+  /**
+   * When user press `enter` key
+   */
+  private _onSubmit(state: { value: unknown }): void {
+    this.status = 'answered';
+    // Rerender prompt (and clean subline error)
+    this.render();
+
+    this.screen.done();
+    this.done(state.value);
+  }
+
+  private _onError(state: inquirer.prompts.FailedPromptStateData): void {
+    this.render(state.isValid || undefined);
+  }
+
+  private _onKeyPress(event: IKeyPressEvent): void {
+    if (event.key.ctrl) {
+      switch (event.key.name) {
+        case 'backspace':
+          return this._setQuery('');
+      }
+    } else {
+      switch (event.key.name) {
+        // Go to beginning of list
+        case 'home':
+          return this._adjustSelected(-Infinity);
+        // Got to end of list
+        case 'end':
+          return this._adjustSelected(Infinity);
+        // Paginate up
+        case 'pageup':
+          return this._adjustSelected(-(this.opt.pageSize ?? 1));
+        // Paginate down
+        case 'pagedown':
+          return this._adjustSelected(this.opt.pageSize ?? 1);
+
+        case 'backspace':
+          return this._setQuery(this._query.slice(0, -1));
+        case 'up':
+          return this._onUpKey();
+        case 'down':
+          return this._onDownKey();
+
+        default:
+          if (event.key.sequence && event.key.sequence.length === 1) {
+            this._setQuery(this._query + event.key.sequence);
+          }
+      }
+    }
+  }
+
+  private _setQuery(query: string): void {
+    this._query = query;
+    const filter: string = query.toUpperCase();
+
+    const { choices } = this.opt.choices;
+    for (const choice of choices as Iterable<{ disabled?: boolean; type: string; short: string }>) {
+      if (choice.type !== 'separator') {
+        choice.disabled = !choice.short.toUpperCase().includes(filter);
+      }
+    }
+
+    // Select the first valid option
+    this._adjustSelected(0);
+  }
+
+  // Provide the delta in deplayed choices and change the selected
+  // index accordingly by the delta in real choices
+  private _adjustSelected(delta: number): void {
+    const { choices } = this.opt.choices;
+    const pointer: number = this._selected;
+    let lastValidIndex: number = pointer;
+
+    // if delta is greater than 0, we are moving down in list w/ selected index
```

Comment:
This comment and the other branch are swapped.
Also, if delta is exactly 0, the request is to adjust to the first displayed choice that has an index >= the current selected choice.